app/src/auth.ts

```diff
@@ -21,31 +23,46 @@ type LoginParams = {
 	share?: boolean;
 };
 
-function getAuthEndpoint(provider?: string, share?: boolean) {
-	if (share) return '/shares/auth';
-	if (provider === DEFAULT_AUTH_PROVIDER) return '/auth/login';
-	return `/auth/login/${provider}`;
-}
-
 export async function login({ credentials, provider, share }: LoginParams): Promise<void> {
-	const response = await api.post<any>(getAuthEndpoint(provider, share), {
-		...credentials,
-		mode: 'session',
-	});
-
-	// Refresh the token 10 seconds before the access token expires. This means the user will stay
-	// logged in without any noticeable hiccups or delays
-
-	// setTimeout breaks with numbers bigger than 32bits. This ensures that we don't try refreshing
-	// for tokens that last > 24 days. Ref #4054
-	if (response.data.data.expires <= 2100000000) {
-		refreshTimeout = setTimeout(() => refresh(), response.data.data.expires - 10000);
-	}
-
 	const appStore = useAppStore();
 	const serverStore = useServerStore();
 
-	appStore.accessTokenExpiry = Date.now() + response.data.data.expires;
+	let response: AuthenticationData;
+
+	if (share) {
+		const { share, password } = credentials;
+		if (!share) throw new Error('Missing share ID');
+
+		await sdk.request(authenticateShare(share, password, 'session'));
+		// To initialize auto-refresh
+		response = await sdk.refresh();
+	} else {
+		const { email, identifier, password, otp } = credentials;
+
+		if (!password) throw new Error('Missing password');
+
+		const loginOptions: LoginOptions = { otp, ...(provider !== DEFAULT_AUTH_PROVIDER && { provider }) };
+
+		if (email) {
+			response = await sdk.login(email, password, loginOptions);
+		} else if (identifier) {
+			const login =
+				<Schema extends object>(): RestCommand<AuthenticationData, Schema> =>
+				() => {
+					const path = getAuthEndpoint(loginOptions.provider);
+					const data = { identifier, password, otp, mode: 'session' };
+					return { path, method: 'POST', body: JSON.stringify(data) };
+				};
+
+			await sdk.request(login());
+			// To initialize auto-refresh
+			response = await sdk.refresh();
```

Comment:
This should just be setting the token and expiry received from the login instead of doing a full refresh call.